langchain/src/tools/gmail/send_message.ts

```diff
@@ -0,0 +1,74 @@
+import { GmailBaseTool, GmailBaseToolParams } from "./base.js";
+
+export interface SendMessageSchema {
+  message: string;
+  to: string | string[];
+  subject: string;
+  cc?: string | string[];
+  bcc?: string | string[];
+}
+
+export class GmailSendMessage extends GmailBaseTool {
+  name = "gmail_send_message";
+
+  description = "Send a message using Gmail";
+
+  constructor(fields?: GmailBaseToolParams) {
+    super(fields);
+  }
+
+  private createEmailMessage({
+    message,
+    to,
+    subject,
+    cc,
+    bcc,
+  }: SendMessageSchema): string {
+    const emailLines: string[] = [];
+
+    // Format the recipient(s)
+    const formatEmailList = (emails: string | string[]): string =>
+      Array.isArray(emails) ? emails.join(",") : emails;
+
+    emailLines.push(`To: ${formatEmailList(to)}`);
+    if (cc) emailLines.push(`Cc: ${formatEmailList(cc)}`);
+    if (bcc) emailLines.push(`Bcc: ${formatEmailList(bcc)}`);
+    emailLines.push(`Subject: ${subject}`);
+    emailLines.push("");
+    emailLines.push(message);
+
+    // Convert the email message to base64url string
+    const email = emailLines.join("\r\n").trim();
+    // this encode may be an issue
+    return Buffer.from(email).toString("base64url");
+  }
+
+  async _call({
```

Comment:
It doesn't seem like `_call` it should be overriding anything but when we run the example file it seems to correctly call the method and make the API request to gmail.
Also we used `Tool` because it extends `StructuredTool` already, and google calendar was also doing the same thing. But on second thought, we aren't overriding the `call` method defined in `Tool` anyways, so maybe that's pointless. We can make the change to make `GmailBaseTool extend StructuredTool` instead.

Reply:
On second thought, it seems like a `Toolkit` requires a list of `Tool` only, not `StructuredTool`. https://github.com/langchain-ai/langchainjs/blob/d82cfb84ac60367595d19e77175c7f1c596c247c/langchain/src/agents/toolkits/base.ts#L8-L10

So making these of type `StructuredTool` would mean we cannot make a Toolkit with the Gmail tools right? In that case is it better to keep these as a `Tool`?

Reply:
Yeah, we can modify Toolkit or I suppose make a `StructuredToolkit` in the future but these should definitely extend `StructureTool`

Reply:
Alright that makes sense. We've made the change to convert `GmailBaseTool` into a `StructuredTool` already